### PR TITLE
Initial System.Text.Encoding.Extensions.Tests cleanup

### PR DESCRIPTION
**Pull request by hughbe**
- UTF7Encoding, UTF32Encoding, ASCIIEncoding

Some tests are duplicates that I didn't delete before - e.g. lone high/low surrogate, single ASCII char or byte

I have left UTF8Encoding and UnicodeEncoding unaddressed at the moment as I need to add support for NegativeEncodingTests.Encode_Invalid and NegativeEncodingTests.Decode_Invalid in UnicodeEncoding, and I wouldn't mind getting some sleep tonight :P

Contributes to #8114

/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.cs [hunk 1]
```diff
@@ -40,14 +41,16 @@ public void Decode(byte[] bytes, int index, int count)
 
         public static IEnumerable<object[]> Decode_InvalidBytes_TestData()
         {
-            // Non ASCII bytes
+            // All non-ASCII Latin1 bytes
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.cs]: nit: just call this Latin-1 Supplement instead of non-ASCII Latine1

**Comment by tarekgh** [comment]
LGTM. good job @hughbe

**Comment by tarekgh** [comment]
test Innerloop Windows_NT Release Build and Test please

File: src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.cs]: Done

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu14.04 Release Build and Test please
Test Innerloop Windows_NT Release Build and Test please

**Comment by tarekgh** [comment]
test Innerloop Ubuntu14.04 Release Build and Test please

**Pull request merged**